### ISSUE TITLE
An unknown error occurred. (Status: 0) Maybe the request timed out?

### ISSUE THREAD

**Issue by Frituurpanda** [amancevice/caravel#13]
"An unknown error occurred. (Status: 0) Maybe the request timed out?"

I'm suddenly getting a lot of these when loading graphs. I know this might be a very generic Caravel related issue. Caravel however seems to work fine without Docker on this system.

Kind regards,

**Comment by amancevice** [comment]
Which tag are you using? Without knowing more about your setup I'm afraid I won't be much help

**Comment by Frituurpanda** [comment]
The error occurs on a complete stock version that was build 2 hours ago. When executing the demo content and opening one of the dashboards like "Births" for example, i'll see the "Status: 0" error displayed in a few slices like "Genders by state", "Title" and the "Name Cloud" one. 

Again, i don't think this might necessarily have something to do with the 'docker implementation' and this might very well be a caravel issue/problem on my server.

**Comment by amancevice** [comment]
So you're building a Docker image from the Dockerfile yourself from this repo? Or are you pulling a tag from Docker hub?

In essence, without divulging sensitive information, what command are you running to start the container?

Caravel needs to be able to communicate with _both_ the Database that's hosting the caravel app itself as well as the database that's hosting the data you wish to plot. It sounds to me like it might be having trouble communicating with one or the other for you.

**Comment by Frituurpanda** [comment]
Just complete stock. 

Only ran:
docker run --name caravel -d -p 8088:8088 amancevice/caravel
docker exec -it caravel demo
for this concept. 

I have the same problem with my version that works with a mysql database. The complete stock version however displays the same problem in the demo section. I'm going to try this on a different machine to see if there might be a problem with this server that is causing this problem.

**Comment by amancevice** [comment]
Running that command will create an ephemeral SQLite db file in the container that hosts both the Caravel DB and the demo DB.

I suppose your server could be inadequate or overloaded but I would think even the simplest of servers could support that. Let me know what you find.

**Comment by amancevice** [comment]
You could also try using an older image tag, like `amancevice/caravel:0.10.0` or `amancevice/caravel:0.9.1`, though I'm not optimistic this would have an effect.

**Comment by Frituurpanda** [comment]
Don't think the server has a problem tbh, it has about 16GB of ram available and a decent cpu.. You can view the problem here: http://imgur.com/a/GvJxH. Did not find any relation at this point as it was working fine a few weeks back.

**Comment by amancevice** [comment]
The `latest` tag hasn't been updated in [21 days](https://hub.docker.com/r/amancevice/caravel/tags/). Have you tried the `0.10.0` tag or `0.9.1`?

**Comment by amancevice** [comment]
you could also try running as root, since I believe that was that last change to the `latest` tag. I'm no sure why this wouldn't work if you are using an ephemeral DB file in the demo like you say, but here's how you would run as root:

``` bash
docker run --name caravel -d -p 8088:8088 --user root amancevice/caravel
docker exec -it caravel demo
```

**Comment by Frituurpanda** [comment]
I'll go and try running it as root, i'll report back on the results.

**Comment by Frituurpanda** [comment]
No succes as you can see here: http://imgur.com/a/LMVLd.. 
Going to try and run the 'normal' caravel installation and see if that changes anything. Thanks for helping me out, appreciate it!

**Comment by amancevice** [comment]
No problem! Sorry for your troubles. I'm going to close this issue because I can't reproduce and don't see a clear fix from my end. If you end up finding a smoking gun feel free to comment here or in a new issue.

**Comment by amancevice** [comment]
@Frituurpanda I started seeing this on an old mac mini I have running next to my desk to display a dashboard. There isn't much in the container logs that point to the host being the issue, so I suspect it's got to do with the mac mini itself. I tried turning off WiFi and relying on a wired connection and that seems to have helped. It could also be the VPN software I'm using to access container (which is inside our VPC). So far turning off WiFi seems to have done the trick
